src/test/kotlin/CrashReaderIntegrationTest.kt

```diff
@@ -1,85 +1,108 @@
-            assert(crash is Crash.Java)
-            assert((crash as Crash.Java).code == expectedCode)
-        }
+    @ParameterizedTest
+    @ResourceFilesSource("crashes/minecraft")
+    fun shouldRunAllMinecraftCrashes(crashFile: File) {
+        val name = crashFile.name
+        val parts = name.split("-")
+        val isModded = parts[0] == "true"
+        val expectedException = parts[1]
+
+        val crashEither = crashReader.processCrash(crashFile.readLines(), tempDir)
+        assertTrue(crashEither is Either.Right)
+        val crash = (crashEither as Either.Right).b
+        assertTrue(crash is Crash.Minecraft)
+        assertEquals(isModded, (crash as Crash.Minecraft).modded)
+        assertTrue(crash.exception.contains(expectedException))
     }
 
-    @Test
-    fun shouldRunAllLogs() {
-        val crashes = File(this::class.java.getResource("crashes/log").file).listFiles()
+    @ParameterizedTest
+    @ResourceFilesSource("crashes/java")
+    fun shouldRunAllJavaCrashes(crashFile: File) {
+        val name = crashFile.name
+        val parts = name.split("-")
+        val expectedCode = parts[0]
 
-        crashes.forEach {
-            val crashEither = crashReader.processCrash(it.readLines(), tempDir)
-            assert(crashEither is Either.Right)
-            val crash = (crashEither as Either.Right).b
-            assert(crash is Crash.LauncherLog)
-        }
+        val crashEither = crashReader.processCrash(crashFile.readLines(), tempDir)
+        assertTrue(crashEither is Either.Right)
+        val crash = (crashEither as Either.Right).b
+        assertTrue(crash is Crash.Java)
+        assertEquals(expectedCode, (crash as Crash.Java).code)
     }
 
-    @Test
-    fun shouldProcessDeobfuscation() {
-        val crashes = File(this::class.java.getResource("crashes/deobfuscator").file).listFiles()
-
-        crashes.forEach {
-            if (!it.nameWithoutExtension.endsWith("deobf")) {
-                val either = crashReader.processCrash(it.readLines(), tempDir)
-                val content = File(it.parent, it.nameWithoutExtension + "-deobf.log")
-                if (content.exists()) {
-                    assertDeobf(either, content.readText())
-                } else {
-                    assertDeobf(either, null)
-                }
-            }
-        }
+    @ParameterizedTest
+    @ResourceFilesSource("crashes/log")
+    fun shouldRunAllLogs(crashFile: File) {
+        val crashEither = crashReader.processCrash(crashFile.readLines(), tempDir)
+        assertTrue(crashEither is Either.Right)
+        val crash = (crashEither as Either.Right).b
+        assertTrue(crash is Crash.LauncherLog)
     }
 
-    private fun assertDeobf(either: Either<ParserError, Crash>, content: String?) {
-        assert(either is Either.Right)
+    /** Removes either a trailing `\r\n`, `\n` or `\r`. */
+    private fun String.dropLastLineTerminator() = removeSuffix("\n").removeSuffix("\r")
```

Comment:
This is necessary because for Windows the linter terminator is `\r\n` and therefore the tests were failing on Windows.